app/src/main/java/de/westnordost/streetcomplete/quests/parking_fee/FeeAnswer.kt

```diff
@@ -1,11 +1,38 @@
 package de.westnordost.streetcomplete.quests.parking_fee
 
+import de.westnordost.streetcomplete.data.meta.updateWithCheckDate
+import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder
 import de.westnordost.streetcomplete.quests.opening_hours.model.OpeningHoursRuleList
 
 
-sealed class FeeAnswer
+sealed class FeeAnswer {
+    abstract fun applyTo(changes: StringMapChangesBuilder)
+}
 
-object HasFee : FeeAnswer()
-object HasNoFee : FeeAnswer()
-data class HasFeeAtHours(val openingHours: OpeningHoursRuleList) : FeeAnswer()
-data class HasFeeExceptAtHours(val openingHours: OpeningHoursRuleList) : FeeAnswer()
+object HasFee : FeeAnswer() {
+    override fun applyTo(changes: StringMapChangesBuilder) {
+        changes.updateWithCheckDate("fee", "yes")
+        changes.deleteIfExists("fee:conditional")
+    }
+}
+
+object HasNoFee : FeeAnswer() {
+    override fun applyTo(changes: StringMapChangesBuilder) {
+        changes.updateWithCheckDate("fee", "no")
+        changes.deleteIfExists("fee:conditional")
+    }
+}
+
+data class HasFeeAtHours(val openingHours: OpeningHoursRuleList) : FeeAnswer() {
+    override fun applyTo(changes: StringMapChangesBuilder) {
+        changes.updateWithCheckDate("fee", "no")
+        changes.addOrModify("fee:conditional", "yes @ (${openingHours})")
+    }
+}
+
+data class HasFeeExceptAtHours(val openingHours: OpeningHoursRuleList) : FeeAnswer() {
+    override fun applyTo(changes: StringMapChangesBuilder) {
+        changes.updateWithCheckDate("fee", "yes")
+        changes.addOrModify("fee:conditional", "no @ (${openingHours})")
+    }
+}
```

Comment:
Actually, it would be better to have the `applyTo` as an extension function (or just static function). To apply an answer to a StringMapChangesbuilder is not an essential part of a fee answer, rather it is a functionality of the quest.